### PR TITLE
Bombsuit and Specialised Armour Tweaks

### PR DESCRIPTION
**Pull request by NanakoAC**
An overhaul of the EOD suit to make it more believable and suitable for purpose, plus a little tweak to riot/bulletproof/ablative suits

  - rscadd: "Bomb suit and hood are now far more robust, and resistant to all types of damage"
  - rscadd: "Bomb suits now protect all bodyparts except the hands"
  - tweak: "Bomb suit slowdown significantly increased"
  - rscadd: "Bomb suits now cause the wearer to gradually overheat and will eventually cause heatstroke, their materials are very bad for dissipating bodyheat"
  - rscadd: "Bomb hoods now restrict peripheral vision like welding goggles, but do not protect your eyes from light"
  - tweak: "Bomb suits and bomb hoods are now too large to fit in a backpack"
  - rscadd: "Bulletproof, ablative and riot suits are no longer cripplingly overspecialised, their resistance to the non-primary damage types has been increased"

### CODE REVIEW

File: code/modules/clothing/suits/utility.dm [hunk 1]
```diff
@@ -47,27 +47,71 @@
...
+		wearer = user
+		wearer << "<span class='Notice'>You struggle into the [src]. It feels hot, heavy and uncomfortable</span>"
+		processing_objects.Add(src)
+
+
+	..(user, slot)
+
+#define 	BOMBSUIT_THERMAL	0.27
+#define		BOMBHOOD_THERMAL	0.12
+#define		BOMBSUIT_MAX_TEMPERATURE	420	//heat 2 for humans, heat 1 for unathi
```
Comment (skull132) [inline code/modules/clothing/suits/utility.dm]: `#undef` all of these at the end of the file, if they are only used within this file.

**Comment by skull132** [comment]
Changelog is a .txt.

**Comment by skull132** [comment]
Also: needs to be removed from the processing objects list upon `Destroy()` and when being unequipped. Otherwise, runtime errors.

**Comment by NanakoAC** [comment]
I'll add the destroy.

It's deliberately not removed from processing when unequipped, because it cools down gradually when you take it off. it removes itself once its done cooling. Was necessary otherwise people would just take it off and put it back on instantly every few minutes to negate the drawback

**Comment by skull132** [comment]
Two Travis errors:
```
code/game/machinery/doors/windowdoor.dm:75:warning: M: variable defined but not used
code/modules/clothing/suits/utility.dm:115:error: destroy: undefined proc
```

Maybe the proc was `Destroyed()`. I forget. Look it up.

**Comment by NanakoAC** [comment]
The windowdoor warning is in development, i think it was in the little bugfix patch the other day

File: code/modules/clothing/suits/utility.dm [hunk 2]
```diff
@@ -47,27 +47,76 @@
...
+/obj/item/clothing/suit/bomb_suit/process()
+	if (!wearer || !istype(loc, /mob))//If nobody's wearing the suit, then it cools down
+		suit_temp -= 0.5
+		if (suit_temp < T20C)
+			suit_temp = T20C
+			processing_objects.Remove(src)
+		return
+
+	var/amount = BOMBSUIT_THERMAL
+	if (istype(wearer.head, /obj/item/clothing/head/bomb_hood))//wearing both parts heats up faster
```
Comment (skull132) [inline code/modules/clothing/suits/utility.dm]: Won't this generate a runtime log in the instance where the suit is carried in hand (at which point, istype(loc, /mob) == TRUE but wearer is still null)?

**Comment by skull132** [comment]
Mkay.

File: code/modules/clothing/suits/utility.dm (hunk 2 above)
Comment (NanakoAC) [inline code/modules/clothing/suits/utility.dm]: damnit, you're right and i'm dumb

will fix

**Comment by LordFowl** [comment]
What exactly is the logic behind removing the "crippling" specialisation of ablative vests and bulletproof vests? Should not the absurd protection against a certain source of damage not have its drawback - absurd weakness against other sources of damage?

**Comment by NanakoAC** [comment]
@LordFowl For one, believability. A thick piece of armoured material is effective, to some degree against everything. It makes little sense that a bulletproof vest would be worse than a smaller officer vest

For two, balance: Having 80% resistance in one category isn't worth glaring weakness, especially when you consider that tactical armour is a thing which exists, and has 60% resistance against melee/bullet/lasers. Right now tac armour is an obvious best choice in all circumstances.

I believe choosing a specialised vest should tradeoff some protection for an increase in other areas, but not entirely strip most of your defenses

**Comment by LordFowl** [comment]
@NanakoAC It hardly "strips you of your defenses" considering that it still provides you more protection than anyone not wearing armour, and a hell lot more protection in a specific area even against those wearing armour. Senselessly buffing other armours is not the solution to Tactical armour. If Tactical armour is an issue the only solution is tackling it at its source.

**Comment by NanakoAC** [comment]
I don't think its senselessly, though.

Killing someone with rapid shots is pretty fast and easy, i don't think 60% reduction is so much that it needs a nerf. Personally i think tactical armour is fine, its these other ones that were the problem. Therefore **have** tackled the problem at what, imo, is the source

**Comment by LordFowl** [comment]
If anything is flawed, it is tactical armour. The specialised armours were perfectly fine before this meddling - sufficiently skewed to provide protection against a similarly specialised enemy, serving its purpose well. Diluting its purpose serves nothing except to make it more powerful, and less of a tactical choice.

**Comment by NanakoAC** [comment]
What's wrong with tactical armor? It won't negate even half of shots, most things are going to hurt.

I edited my comment a few times with numbers and notes

The thing to realise here, is that the effectiveness of armour scales up exponentially with the armour value.  Tactical armour is pretty balanced, and is at about a sweet spot of being just right.

The 80 armour value on the specialised armours, towards their specialty, isn't just 20% higher than tactical armour. IT brings them up to a 64% negation chance for their special type, which is a little under double what tac armour does, that extra amount goes a long way. Conversely, tac armour's 60% is still far, far more powerful than the 25% towards lesser types i've given here.  10% is frankly worthless and may as well not exist, it's not believable that wearing a big chunk of matter on your chest would do nothing against shots that aren't of a specific type

**Comment by NanakoAC** [comment]
formula for damage reduction:
where armor = the armor value as a percentage (so divide it by 100)
1 -((1 - armor) + ((armor * (1 - armor))*0.5))

10% armor = 5.5% reduction
20% armor = 12% reduction
30% armor = 19.5% reduction
40% armor = 28% reduction
50% armor = 37.5% reduction
60% armor = 48% reduction
70% armor = 59.5% reduction
80% armor = 72% reduction
90% armor = 85.5% reduction
100% armor = 100% reduction

**Comment by OneOneThreeEight** [comment]
@NanakoAC Is this the actual armor resistance system Zuhayr was talking about over at Bay some months ago? Or are we still on the vanilla version of Halfblock/Fullblock?

**Comment by skull132** [comment]
Still vanilla, @OneOneThreeEight

**Comment by Serveris** [comment]
Last I checked, EOD suits were completely useless for dealing with or even being remotely near bombs due to them being coded as armor. 

The issue here being, sure, you might survive a direct blast, but then you're left dying from pressure not even two seconds later due to the gaping hole you're now standing in. Was this taken into account? If not, then this is really not going to make it more applicable. It's never used for this exact reason.

**Comment by NanakoAC** [comment]
@Serveris I added pressure resistance to it, same as the firesuit. If you wear an oxygen mask with it, it will be EVA capable.

**Comment by skull132** [comment]
Has immense amount of slowdown. I'm fine with testing this.

**Comment by OneOneThreeEight** [comment]
We Bulldozer now.

**Pull request merged**